paint/support/src/dispatch.rs

```diff
@@ -1009,9 +1050,19 @@ macro_rules! decl_module {
 		$vis fn $name(
 			$origin: $origin_ty $(, $param: $param_ty )*
 		) -> $crate::dispatch::DispatchResult<$error_type> {
-			{ $( $impl )* }
-			// May be unreachable.
-			Ok(())
+			#[cfg(feature = "std")]
+			{
+				use $crate::tracing;
+				let span = tracing::span!(tracing::Level::INFO, stringify!($name));
+				let _enter = span.enter();
+				let result: $crate::dispatch::DispatchResult<$error_type> = (move || { { $( $impl )* } Ok(()) })();
+				result
+			}
+			#[cfg(not(feature = "std"))]
+			return {
```

Comment:
```suggestion
		 {
```

I think you don't need the returns + no need to store 'result' everywhere

Reply:
good catch - I had just copied old code without thinking enough